### ISSUE TITLE
One slow client can reduce the image quality for all other users

### ISSUE THREAD

**Issue by tdjac0bs** [bigbluebutton/bigbluebutton#12274]
I've noticed the drop of video quality in a situation that I was expecting cristal clear recordings ( broadcaster in same network as BigBlueButton server ).

Discussing with @prlanzarin he pointed that a slow client ( 3g connected ) can send packets telling that it's not able to follow the stream, causing the broadcaster to reduce it's quality.

This issue is to cover this topic ( in case of other users are experiencing same problem ) and discuss possible solutions.

To reproduce:
1. Connect with a full speed user ( no BW limits ) with name "Broadcaster" and share webcam ;
2. Connect with a full speed user ( no BW limits ) with name "Viewer" and just stay on room ;
3. At this point you'll see a good quality image.
4. Connect with a browser limited to 1mbit with name "Limited viewer" ;
5. From this point both "Viewer" and "Limited viewer" will have decreased video quality ;

Expected behaviour:
1. The quality should not drop for clients with good network, despite of users having bad network ( this way one student with very bad connection is affecting quality of the whole class ).

**Comment by hostbbb** [comment]
prime case for getting a media engine with simulcast as option to kurento.  The SFU can then set the low quality to stream to slow clients. and preserve the best quality for recording, and other capable clients.

**Comment by Jossef-767** [comment]
Is there a way (configuration) to prevent the slow client from sending a message that it is unable to follow the stream.?
Or make kurento, Or SFU, to ignore this message from the slow client?

**Comment by prlanzarin** [comment]
No. It works like that by design and there is no configuration available to disable congestion control.

**Comment by prlanzarin** [comment]
> No. It works like that by design and there is no configuration available to disable congestion control.

Following up on this comment: REMB (congestion control) is propagated E2E by design. **I have no intention of disabling it**, nor do I have any intention of spending time making Kurento terminate REMB on the receiver end or to fix the REMB control parameters (which are kind of broken).

If it's up to me there should be no action on this over the current media server. There will be action on this once we have a client facing media server that does simulcast and TWCC instead of REMB. One that also has NACK buffers and allows control over whether twcc is relevant only for inbound streams. 

The current approach works like this due to the fact that the lack of the aforementioned characteristics make it a viable design if one is trying to encompass the largest amount of network scenarios (which _we are_). What we aren't trying to provide as of 2.3 (and we don't advertise to be trying AFAIK) is having bitrate-stable Full HD streams.

**Comment by JDFDEV** [comment]
I am guessing the problem here is how often this occurs... In Countries that still are lagging behind with good bandwidth coverage in Rural areas (and nowadays thanks to the marvels of Home Office a lot of users are confined to those lousy connections) you often get in Webinars that the connection brakes down due to some bad connection... 

Ok. Now in BBB 2.3 at least I can see who is the one responsible thanks to network meter... and i can decide what to do with him :D 

Joking aside, I think the ability to disable this Remb behavoir shold be somehow doable... (or not?)
https://doc-kurento.readthedocs.io/en/latest/_static/client-javadoc/org/kurento/client/BaseRtpEndpoint.html

**Comment by prlanzarin** [comment]
> you often get in Webinars that the connection brakes down due to some bad connection...

I'm not sure what's the meaning of "breaking down", but I'll interpret it literally as in "cease to function". A sender connection **will not break down** due to a receiver end network outage. Never. It'll degrade, but it **will not break down**.

> Joking aside, I think the ability to disable this Remb behavoir shold be somehow doable... (or not?)

As I said in my previous comment:

> to fix the REMB control parameters (which are kind of broken).

I'll add the option to disable REMB via SDP munging in bbb-webrtc-sfu in a future 2.3 release so at least people have something to work with. Other than that, refer to https://github.com/bigbluebutton/bigbluebutton/issues/12274#issuecomment-834480879.

**Comment by JDFDEV** [comment]
What a low latency response that was! 😄👍 

You are absolutely right.. It doesn't "brake down".. but the TX/RX rates go down the basement. And unfortunately it happens more often these days...

That OFF-Button would at least be worth a try... Thanks in advance!

**Comment by prlanzarin** [comment]
> That OFF-Button would at least be worth a try... Thanks in advance!

https://github.com/bigbluebutton/bbb-webrtc-sfu/blob/1f82c241a6383c811c73871224e6f4670b5e45c4/config/default.example.yml#L187-L188

It won't stop things from being degraded due to relayed packet loss info and etc, but the REMB part specifically can now be deactivated. That's it for the short term.
